src/main/java/org/jboss/modules/ModuleLoader.java

```diff
@@ -574,13 +575,33 @@ protected final Module findLoadedModuleLocal(String name) {
      * @param module the module to unload
      * @throws SecurityException if an attempt is made to unload a module which does not belong to this module loader
      * @throws SecurityException if the module was not defined by this module loader
+     * @deprecated use {@link #unloadModuleLocal(String,Module)} instead
      */
+    @Deprecated
     protected final void unloadModuleLocal(Module module) throws SecurityException {
+        unloadModuleLocal(null, module);
```

Comment:
OK, I see the reason to keep unloadModuleLocal(String, Module) with two args. 

However, if the caller is supposed to know which name-module association should be removed, why the devious step to `null` ? I vote for requiring a non-null `moduleName` in unloadModuleLocal(String, Module). In that way, we can avoid the call to the deprecated `Module.getIdentifier()` and make the execution path more straightforward.